src/Plugin/ElasticsearchIndexBase.php

```diff
@@ -106,11 +106,26 @@ protected function getEventDispatcher() {
   }
 
   /**
-   * @inheritdoc
+   * {@inheritdoc}
    */
   public function setup() {
-    // TODO: create index templates.
-    // $this->client->indices()->putTemplate().
+    // Create an index if index definition is provided by the index plugin.
+    if ($index_definition = $this->getIndexDefinition()) {
+      $index_name = $this->getIndexName([]);
+
+      if (!$this->client->indices()->exists(['index' => $index_name])) {
+        $this->client->indices()->create([
+          'index' => $index_name,
+          'body' => $index_definition->toArray(),
+        ]);
+      }
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getIndexDefinition() {
```

Comment:
Maybe this empty method could be removed, then use a conditional check for the method from `setup()`?

`if (method_exists($this, 'getIndexDefinition') && $index_definition = $this->getIndexDefinition()) {`

Reply:
Method `getIndexDefinition()` is defined in `ElasticsearchIndexInterface` interface, therefore it is implemented in `ElasticsearchIndexBase`.